libs/model/src/featurelayer.cpp

```diff
@@ -83,11 +83,127 @@ TileFeatureLayer::TileFeatureLayer(
 
 TileFeatureLayer::~TileFeatureLayer() = default;
 
+/*
+ * Checks if featureIdParts match one of the uniqueIdComposition specifications.
+ * The order of values in KeyValuePairs must be the same as in the composition!
+ */
+bool TileFeatureLayer::validFeatureId(
+    FeatureTypeInfo const& typeId,
+    KeyValuePairs const& featureIdParts,
+    bool excludeTilePrefix) {
+
+    for (auto& candidateComposition : typeId.uniqueIdCompositions_) {
+        auto compositionPart = candidateComposition.begin();
+
+        if (excludeTilePrefix && this->featureIdPrefix().has_value()) {
+
+            // Iterate past the prefix in the unique id composition.
+            auto featureIdPrefixSize = this->featureIdPrefix().value()->size();
+
+            if (featureIdPrefixSize >= candidateComposition.size()) {
+                // TODO check and raise error when the tile prefix is set?
+                throw logRuntimeError("Tile prefix is longer than a composition.");
+            }
+            while (featureIdPrefixSize > 0) {
+                ++compositionPart;
```

Comment:
It is important to also validate the id parts that are pulled in through the prefix.